app/os_windows.go

```diff
@@ -954,4 +1007,132 @@ func configForDPI(dpi int) unit.Metric {
+		a new window. That behavior doesn't align with iOS/Android/macOS, where the deeplink
+		sends the event to the running app (if any).
+
+		In order to align the behavior, we use a global mutex to check if the app is already
+		running. If it is, we send the deeplink to the running app and exit the new instance.
+
+		That should happen on init to prevent the user from seeing the new window.
+	*/
+	if ok := processDeeplink(); ok {
+		os.Exit(0)
+		return
+	}
+
+	for _, scheme := range schemesDeeplinkList {
+		go registerDeeplink(scheme)
+	}
+}
+
+func registerDeeplink(scheme string) error {
+	path, err := os.Executable()
+	if err != nil {
+		return err
+	}
+
+	key, _, err := registry.CreateKey(registry.CURRENT_USER, `Software\\Classes\\`+scheme, registry.ALL_ACCESS)
+	if err != nil {
+		return err
+	}
+	defer key.Close()
+
+	if err = key.SetStringValue("", "URL:"+scheme+" Protocol"); err != nil {
+		return err
+	}
+
+	if err = key.SetStringValue("URL Protocol", ""); err != nil {
+		return err
+	}
+
+	icon, _, err := registry.CreateKey(key, `DefaultIcon`, registry.ALL_ACCESS)
+	if err != nil {
+		return err
+	}
+	defer icon.Close()
+
+	if err = icon.SetStringValue("", `"`+path+`",1`); err != nil {
+		return err
+	}
+
+	cmd, _, err := registry.CreateKey(key, `shell\\open\\command`, registry.ALL_ACCESS)
+	if err != nil {
+		return err
+	}
+	defer cmd.Close()
+
+	// The app will launch with `-gio_deeplinking` as the first argument, and the deeplink as the second.
+	if err = cmd.SetStringValue("", `"`+path+`" -gio_deeplinking "%1"`); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func processDeeplink() (finishProcess bool) {
+	_, alreadyExists, err := windows.CreateMutex(schemesDeeplink)
+	if err != nil {
+		return false
+	}
+
+	if len(os.Args) < 3 || os.Args[1] != "-gio_deeplinking" {
+		// Only one instance of the app can run at a time.
+		// If the current instance doesn't have any deeplink to process, then
+		// it will exit if another instance is already running, or it will
+		// continue if it is the first instance.
+		return alreadyExists
+	}
+
+	u := os.Args[2]
+
+	if !alreadyExists {
+		// The mutex was created by this process, and we are the first instance.
+		startupDeeplink = u
+		return alreadyExists
+	}
+
+	id := make([]byte, unsafe.Sizeof(int(0)))
+	rand.Read(id)
+
+	// First, I tried to use mmap, but that plagues everything with unsafe pointers, and it's not worth it.
+	f, err := os.Create(filepath.Join(os.TempDir(), "gio_deeplinking-"+hex.EncodeToString(id)))
+	if err != nil {
+		return alreadyExists
+	}
+
+	if _, err := f.Write([]byte(u)); err != nil {
+		return alreadyExists
+	}
+
+	if err := f.Close(); err != nil {
+		return alreadyExists
+	}
+
+	if err := windows.SendMessage(windows.HWND_BROADCAST, windows.GIO_DEEPLINKING, uintptr(len(u)), *(*uintptr)(unsafe.Pointer(&id[0]))); err != nil {
+		return alreadyExists
+	}
+
+	return alreadyExists
```

Comment:
I think that doesn't feels right to add it in gio-core. However, doesn't make sense to left Windows with complete different behaviour.

Reply:
I take a look at Flutter's implementation (https://pub.dev/packages/app_links), ironically they enforce the same limitation:

```
  // Dispatches link if any.
  // This method enables our app to be with a single instance too.
  // This is mandatory if you want to catch further links in same app.
  bool SendAppLinkToInstance(const std::wstring& title);
  ```

So, "enables our app to be with a single instance too" is required.  However, in the source-code they use WM_COPYDATA, instead of a custom event, and create a code that goes inside WM_COPYDATA. Maybe it's better? If I change it, then it will be simplified.

Reply:
If `WM_COPYDATA` simplifies this, let's do that :) In either case, let's do the "easy" platforms first, then add Windows later.

Reply:
I tried to simplify the Windows implementation.